src/components/WalletDropdown/MiniPortfolio/Activity/parseLocal.ts

```diff
@@ -182,24 +181,22 @@ export function parseLocalActivity(
   return { ...defaultFields, ...additionalFields }
 }
 
-export function useLocalActivities(): ActivityMap | undefined {
+export function useLocalActivities(account: string): ActivityMap {
   const allTransactions = useMultichainTransactions()
```

Comment:
why aren't we storing local transactions in redux on a per-address basis? is there any use case for combining transactions from all addresses? it could be another potential fix that doesn't require manual filtering.

Reply:
I'm not sure why we store them this way, but I don't want to modify the redux tx flow at all until the backend is able to serve l2 tx's, since localStorage is currently our users only source of l2 tx's and I imagine changing how tx's are stored would result in client browsers loosing reference to that data